phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaSynthesisProcessor.java

```diff
@@ -0,0 +1,261 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.schema;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import org.apache.hadoop.hbase.util.Pair;
+import org.apache.phoenix.parse.AddColumnStatement;
+import org.apache.phoenix.parse.BindableStatement;
+import org.apache.phoenix.parse.ColumnDef;
+import org.apache.phoenix.parse.ColumnName;
+import org.apache.phoenix.parse.CreateIndexStatement;
+import org.apache.phoenix.parse.CreateTableStatement;
+import org.apache.phoenix.parse.DropColumnStatement;
+import org.apache.phoenix.parse.SQLParser;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SchemaSynthesisProcessor extends SchemaProcessor {
+    public static final String
+            ENTITY_NAME_IN_BASE_AND_ALTER_DDL_DON_T_MATCH =
+            "Entity name in base and alter DDL don't match";
+    private final String baseDDLFile;
+    private final String alterDDLFile;
+
+    public SchemaSynthesisProcessor(String baseDDLFile, String alterDDLFile) {
+        this.baseDDLFile = baseDDLFile;
+        this.alterDDLFile = alterDDLFile;
+    }
+
+    @Override
+    public String process() throws Exception {
+        String baseDDL = getQueriesFromFile(baseDDLFile).get(0);
+        List <String> alterDDL = getQueriesFromFile(alterDDLFile);
+        for(String s : alterDDL) {
+            baseDDL = synthesize(baseDDL, s);
+        }
+        return baseDDL;
+    }
+
+    private String synthesize(String baseDDL, String alterDDL) throws Exception {
+        BindableStatement createStatement = new SQLParser(baseDDL).parseStatement();
+        BindableStatement alterStatement = new SQLParser(alterDDL).parseStatement();
+        if (createStatement instanceof CreateTableStatement) {
+            CreateTableStatement newCreateStmt = null;
+            CreateTableStatement createStmt = (CreateTableStatement) createStatement;
+            if (alterStatement instanceof AddColumnStatement) {
+                newCreateStmt =
+                        getCreateTableStatement((AddColumnStatement) alterStatement, createStmt);
+            } else if(alterStatement instanceof DropColumnStatement) {
+                newCreateStmt =
+                        getCreateTableStatement((DropColumnStatement) alterStatement, createStmt);
+            }
+            return getCreateTableSQL(newCreateStmt);
+        } else if (createStatement instanceof CreateIndexStatement) {
+            CreateIndexStatement newCreateIndexStmt =
+                    getCreateIndexStatement((CreateIndexStatement) createStatement, alterStatement);
+            return getCreateIndexSQL(newCreateIndexStmt);
+        }
+        return "";
```

Comment:
What about CreateSequenceStatement and CreateFunctionStatement? And if none of the above, shouldn't we get an error of some kind?

Reply:
@swaroopak, This is a great start. As @gjacoby126 pointed out, there are more cases to cover. I think we need a design doc on this to explain how the design covers all the cases and explain how this tool will be used. I also think we can add add a future to Phoenix to self verify/test DDL statements. The place to add such a feature would be MetaDataClient. After every DDL, we can verify if DDL is correctly implemented by Phoenix.

Reply:
Thank you @kadirozde. 
I started the doc per your suggestion. Meanwhile, do you think if this PR can be reviewed/merged for the initial scenarios (would be easier on reviews and incremental addition of code)? Or do you recommend not to merge until all of that is implemented? Please let me know.

Reply:
As far as I understand, this PR significantly refactors an existing tool. However, it does not cover some use cases. If so, refactoring should not reduce the functionality. So, I suggest a quick design review and a complete refactoring.

Reply:
This PR adds a new feature in SchemaTool with SYNTH mode. I added the design doc in the Jira with the current implementation and future extensions. 
As per offline discussions, I'll change the input format from 2 files to 1 file. Thanks, @kadirozde